docs/src/content/concepts-modes.md

```diff
@@ -138,6 +139,62 @@ most cases, the configuration will look like this:
 
 {{< figure src="/schematics/proxy-modes-transparent-3.png" >}}
 
+## WireGuard (transparent proxy)
+
+```shell
+mitmdump --mode wireguard
+```
+
+The WireGuard mode works in the same way as transparent mode, except that setup
+and routing client traffic to mitmproxy are different. In this mode, mitmproxy
+runs an internal WireGuard server, which devices can be connected to by using
+standard WireGuard clients applications, and just by importing the WireGuard
+configuration provided by mitmproxy.
+
+No additional routing configuration is required for this mode, since WireGuard
+operates by exchanging UDP packets between client and server (with encrypted IP
+packets as their payload) instead of routing IP packets directly. Additionally,
+the WireGuard server runs entirely in userspace, so no administrative privileges
+are necessary for setup or operation of mitmproxy in this mode.
+
+### Configuration
+
+By default, the WireGuard server will listen on port `51820/udp`, the default
+port for WireGuard servers, which should not require modifying firewall settings
+in most circumstances. However, changing firewall settings might be necessary
+when specifying a different port (i.e. when using something like
+`--mode wireguard@51821`).
+
+The encryption keys for WireGuard connections are stored in a file on disk, and
+it is possible to specify its path manually with `--mode wireguard:path`. If no
+path is specified, a fallback path within the default mitmproxy configuration
+directory is used. If no file exists at that path when the WireGuard server is
+instantiated, new keys will be generated and stored there automatically.
```

Comment:
```suggestion
The encryption keys for WireGuard connections are stored in `~/.mitmproxy/wireguard.conf`. It is possible to specify a custom path manually with `--mode wireguard:path`. New keys will be generated automatically if the specified file does not exist. For example, to connect two clients simultaneously, you can run `mitmdump --mode wireguard:wg-keys-1.conf --mode wireguard:wg-keys-2.conf@51821`.
```

Reply:
I still think that setting up multiple peers for one WireGuard server would be better than running multiple WireGuard servers, but for now, this makes sense.